src/Uno.UI/UI/Xaml/Controls/ListViewBase/VirtualizingPanelLayout.Android.cs

```diff
@@ -1585,7 +1585,8 @@ private void ResetHeaderAndFooter(RecyclerView.Recycler recycler)
 				var headerIndex = GetHeaderViewIndex();
 				_previousHeaderExtent = GetChildExtentWithMargins(headerIndex);
 				// Rebind to apply changes, RecyclerView alone will recycle the view without rebinding.
-				recycler.BindViewToPosition(GetChildAt(headerIndex), headerIndex);
+				// Here we use position: 0 because the header is always at index 0 from the collection's perspective.
+				recycler.BindViewToPosition(GetChildAt(headerIndex), position: 0);
```

Comment:
headerIndex wasn't 0 in some cases? causing the bug?

Reply:
No, the problem was with the footer. It was using values above 0 or 1.